src/modules/shared/components/HeaderPixwaySDK/components/NavigationLoginPixwaySDK/NavigationLoginPixwaySDK.tsx

```diff
@@ -0,0 +1,72 @@
+import { useState } from 'react';
+
+import { usePixwaySession } from '../../../../hooks/usePixwaySession';
+import { NavigationLoginLoggedButton } from './components/NavigationLoginLoggedButton';
+import { NavigationLoginLoggedButtonMobile } from './components/NavigationLoginLoggedButtonMobile';
+import { NavigationLoginNonLoggedButton } from './components/NavigationLoginNonLoggedButton';
+
+interface NavigationLoginPixwaySDKProps {
+  marginLeft?: number;
+  signInRouter?: string;
+  signUpRouter?: string;
+  toggleLoginMenu?: () => void;
+  loginMenu?: boolean;
+}
+
+export const NavigationLoginPixwaySDK = ({
+  marginLeft = 40,
+  signInRouter,
+  signUpRouter,
+  toggleLoginMenu,
+  loginMenu,
+}: NavigationLoginPixwaySDKProps) => {
+  const { data: session } = usePixwaySession();
+  const [userMenu, setUserMenu] = useState<boolean>(false);
+
+  const toggleTabsMemo = () => {
+    if (toggleLoginMenu) {
+      toggleLoginMenu();
+    } else {
+      setUserMenu(!userMenu);
+    }
+  };
+
+  const validatorOpened = loginMenu ? loginMenu : userMenu;
+
+  const InfoToShow = () => {
+    if (session) {
+      return <NavigationLoginLoggedButton />;
+    } else {
+      return (
+        <NavigationLoginNonLoggedButton
+          signInRoute={signInRouter}
+          signUpRoute={signUpRouter}
+        />
+      );
+    }
+  };
+
+  const InfoToShowMobile = () => {
+    return (
+      <NavigationLoginLoggedButtonMobile
+        menuOpened={validatorOpened}
+        toggleMenu={toggleTabsMemo}
+      />
+    );
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  };
+
+  return (
+    <div>
+      <div
+        style={{ marginLeft: marginLeft + 'px' }}
+        className="sm:pw-flex pw-border-l pw-border-[#777E8F] pw-items-center pw-hidden"
+      >
+        <InfoToShow />
```

Comment:
Se o componente for declarado dentro de outro componente essa maneira de renderizar o componente pode levar a bugs, pois a função de referencia do componente é criada a cada renderização

(Já se perguntou se tem alguma diferença entre renderizar o componente como \<InfoToShow/> e {renderInfo()} ?)
O que é o JSX?

Reply:
Segundo o link que voce me mandou pelo que eu li chamar como função que pode dar problemas... devido aos hooks, que o ideal é declarar como render... como tag

`First off, I'll just tell you the solution:

<div>{items.map(Counter)}</div> -- errado
<div>{items.map(i => <Counter key={i.id} />)}</div> --- certo`

https://kentcdodds.com/blog/dont-call-a-react-function-component

Reply:
Em todos os exemplos desse link os dois componentes são declarados em "nivel global", no caso do código o InfoToShow é declarado dentro de outro componente

Reply:
É que é um anti pattern declarar um componente dentro de outro componente, pois o lifecicle de criação é executado a cada renderização do componente pai.
Aqui tem um artigo detalhando um possivel bug que pode ocorrer, onde os estados ficam sendo resetados: https://levelup.gitconnected.com/code-review-avoid-declaring-react-component-inside-parent-component-1768a645f523

Pra chamar com a sintaxe de tag o correto era a declaração ser feito fora do componente (o componente então só seria declarado uma vez)